src/casp/services/api/services/consensus_engine/strategies.py

```diff
@@ -0,0 +1,230 @@
+        self.normalize = normalize
+        self.prune_threshold = prune_threshold
+
+    def _get_ontology_aware_neighborhood_context(
+        self,
+        neighbors: t.List[MatchResult.Neighbor],
+        neighbors_metadata_dict: t.Dict[str, schemas.CellariumCellMetadata],
+    ) -> t.List[schemas.NeighborhoodSummaryOntologyAware]:
+        """
+        Utilize the ontology-aware method to assign weights to neighbor cells based on their distance and cell type
+        ontology, to inform context summarization.
+
+        :param neighbors: Neighbors of the query cell as determined by the matching engine.
+        :param neighbors_metadata_dict: Metadata for each neighbor cell.
+        :return: A list of `schemas.AnnotationInfoOntologyAware` instances, each representing the weighted cell type
+             ontology term for a neighbor cell.
+        """
+
+        neighbor_distances = np.asarray([neighbor.distance for neighbor in neighbors])
+
+        neighbor_metadata = [neighbors_metadata_dict[neighbor.cas_cell_index] for neighbor in neighbors]
+
+        # Get weights for each neighbor
+        gamma = 1.0 / np.median(neighbor_distances)
+        weights = np.exp(gamma * neighbor_distances)
+
+        total_weight = 0
+        total_usable_neighbors = 0
+        scores_dict = {k: 0 for k in self.cell_ontology_resource.ancestors_dictionary.keys()}
+
+        for neighbor_metadata, weight in zip(neighbor_metadata, weights):
+            # Cell Ontology IDs have the formats: CL:0000000 (in our database) and CL_0000000 (in the ontology graph)
+            neighbor_cell_type_ontology_id = neighbor_metadata.cell_type_ontology_term_id.replace(":", "_")
+
+            total_weight += weight
+            total_usable_neighbors += 1
+
+            # Add weight to the neighbor's ontology ID
+            scores_dict[neighbor_cell_type_ontology_id] += weight
+
+            # Propagate the weight to all ancestors (consistent subgraph of the cell type ontology)
+            ancestor_ontology_ids = self.cell_ontology_resource.ancestors_dictionary[neighbor_cell_type_ontology_id]
+
+            for ancestor in ancestor_ontology_ids:
+                scores_dict[ancestor] += weight
+
+        if self.normalize:
+            scores_dict = {k: v / total_weight for k, v in scores_dict.items()}
+
+        if self.prune_threshold > 0:
+            scores_dict = {k: v for k, v in scores_dict.items() if v >= self.prune_threshold}
+
+        return [
+            schemas.NeighborhoodSummaryOntologyAware(
+                score=score,
+                cell_type_ontology_term_id=cell_type_ontology_term_id,
+                cell_type=self.cell_ontology_resource.ontology_term_id_to_name_dict[cell_type_ontology_term_id],
+            )
+            for cell_type_ontology_term_id, score in scores_dict.items()
+        ]
+
+    def summarize(self, query_ids: t.List[str], knn_query: MatchResult) -> schemas.QueryAnnotationOntologyAwareType:
```

Comment:
expose `prune_threshold` and `distance_exponential_weighting_prefactor` here.

Reply:
use kwargs if you want to keep the base class simple. in that case, put the default values here and not in the inner function.